Kernel_23/include/CGAL/Kernel/function_objects.h

```diff
@@ -3540,29 +3520,14 @@ namespace CommonKernelFunctors {
   {
     typedef typename K::Plane_3     Plane_3;
   public:
-    template<typename>
-    struct result;
-
-    template<typename F, typename A, typename B>
-    struct result<F(A, B)> {
-      typedef typename Intersection_traits<K, A, B>::result_type type;
-    };
-
-    template<typename F>
-    struct result<F(Plane_3, Plane_3, Plane_3)> {
-      typedef boost::optional<
-        boost::variant< typename K::Point_3,
-                        typename K::Line_3,
-                        typename K::Plane_3 > > type;
-    };
 
     // n possibilities, so I keep the template.
     template <class T1, class T2>
-    typename cpp11::result_of< Intersect_3(T1, T2) >::type
+    decltype(auto)
     operator()(const T1& t1, const T2& t2) const
```

Comment:
the typedef Intersect_3 is now left unused and there's a warning : https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.3-Ic-58/AABB_tree/TestReport_lrineau_Ubuntu-Latest.gz